blocks/progress-bar-table/progress-bar-table.js

```diff
@@ -0,0 +1,85 @@
+function calculateRatioPost(elements) {
+  const sizel = elements.length;
+  let ratio1;
+  let ratio2;
+  if (sizel === 2) return [100, 0];
+  const childNodes1 = elements[1].childNodes;
+  const childNodes2 = elements[2].childNodes;
+  ratio1 = parseFloat(childNodes1[7].textContent.substring(0, childNodes1[7].textContent.indexOf(' ')));
+  ratio2 = parseFloat(childNodes2[7].textContent.substring(0, childNodes1[7].textContent.indexOf(' ')));
+  if (ratio2 > ratio1) {
+    ratio1 = Math.floor((ratio1 * 100) / ratio2);
+    ratio2 = 100;
+  } else {
+    ratio2 = Math.floor((ratio2 * 100) / ratio1);
+    ratio1 = 100;
+  }
+  return [ratio1, ratio2];
+}
+
+const callback = (entries) => {
+  entries.forEach((entry) => {
+    const elements = entry.target.querySelectorAll('.progress-bar-table > div');
+    const compareTheValues = calculateRatioPost(elements);
+    const toChangea = entry.target.querySelector('.progress-value-1');
+    toChangea.style.width = ('0%');
+    toChangea.style.width = (`${compareTheValues[0]}%`);
+    const toChangeb = entry.target.querySelector('.progress-value-2');
+    toChangeb.style.width = ('0%');
+    toChangeb.style.width = (`${compareTheValues[1]}%`);
+    if (entries[0].intersectionRatio === 0) {
+      toChangea.style.width = ('0%');
+      toChangeb.style.width = ('0%');
+    }
+  });
+};
+
+function addDynamic2(block) {
+  const options = {
+    root: document.root,
+    rootMargin: '0px',
+    threshold: [0, 1],
+  };
+
+  const observer = new IntersectionObserver(callback, options);
+  const target = block;
+  observer.observe(target);
+}
+
+function buildProgressBarTable(block) {
```

Comment:
- use spacing for readability
- use guard clause
- speaking variable names
- use `forEach` which provides index

for example:
```
function buildProgressBarTable(block) {
  const elements = block.querySelectorAll('.progress-bar-table > div');

  elements.forEach((element, i) => {
    if (i === 0) {
      elements[i].style.display = 'none';
      return;
    }

    element.classList.add('progress-wrapper');

    const progressChildNodes = element.childNodes;
    progressChildNodes[1].classList.add(`progress-title-${i}`);
    progressChildNodes[3].classList.add(`progress-description-${i}`);

    const ratio = progressChildNodes[5].textContent;
    progressChildNodes[5].textContent = '';
    progressChildNodes[5].classList.add('progress');
    progressChildNodes[5].classList.add(`progress-aside-1${i}`);
    progressChildNodes[5].classList.add('aside');

    const progress2 = document.createElement('div');
    progress2.classList.add(`progress-value-${i}`);
    progress2.style.width = ('0%');
    progressChildNodes[5].appendChild(progress2);
    progressChildNodes[7].classList.add('progress');
    progressChildNodes[7].classList.add(`progress-aside-2${i}`);
    progressChildNodes[7].classList.add('aside');

    const progressValue = document.createElement('h2');
    const unitMeasure = progressChildNodes[7].textContent;
    progressChildNodes[7].textContent = '';
    progressChildNodes[7].appendChild(progressValue);
    progressValue.textContent = `${ratio} ${unitMeasure}`;
    progressChildNodes[9].classList.add(`progress-subtitle-${i}`);
  });
}
```